agent/reconciler/host_reconciler.go

```diff
@@ -213,7 +285,24 @@ func (r *HostReconciler) hostCleanUp(ctx context.Context, byoHost *infrastructur
 		if err != nil {
 			return err
 		}
-		if r.SkipK8sInstallation {
+		if r.UseInstallerController {
+			if byoHost.Spec.UninstallationScript == nil {
+				return fmt.Errorf("UninstallationScript not found in Byohost %s", byoHost.Name)
+			}
+			logger.Info("Executing Uninstall script")
+			uninstallScript := *byoHost.Spec.UninstallationScript
+			uninstallScript, err = r.parseInstallationScript(ctx, uninstallScript)
+			if err != nil {
+				logger.Error(err, "error parsing Uninstallation script")
+				return err
+			}
+			err = r.executeScript(ctx, uninstallScript)
+			if err != nil {
+				logger.Error(err, "error execting Uninstallation script")
+				r.Recorder.Event(byoHost, corev1.EventTypeWarning, "UninstallScriptExecutionFailed", "uninstall script execution failed")
+				return err
+			}
+		} else if r.SkipK8sInstallation {
```

Comment:
`r.SkipK8sInstallation` should be the first condition. Because if this is set to true, doesn't matter if we are using the installer controller or not